addons/point_of_sale/models/uom.py

```diff
@@ -1,31 +1,34 @@
 from odoo import api, fields, models
 
 
-class UomCategory(models.Model):
-    _name = 'uom.category'
-    _inherit = ['uom.category', 'pos.load.mixin']
+class UomUom(models.Model):
+    _name = 'uom.uom'
+    _inherit = ['uom.uom', 'pos.load.mixin']
 
-    is_pos_groupable = fields.Boolean(string='Group Products in POS',
-        help="Check if you want to group products of this category in point of sale orders")
+    is_pos_groupable = fields.Boolean(string='Group Products in POS', compute='_compute_is_pos_groupable', store=True,
+        help="Check if you want to group products of this unit in point of sale orders")
 
-    @api.model
-    def _load_pos_data_domain(self, data):
-        return [('id', 'in', [uom['category_id'] for uom in data['uom.uom']])]
+    @api.depends('relative_uom_id')
+    def _compute_is_pos_groupable(self):
+        for uom in self:
+            uom.is_pos_groupable = uom.relative_uom_id.is_pos_groupable if uom.relative_uom_id else False
 
     @api.model
     def _load_pos_data_fields(self, config_id):
-        return ['id', 'name', 'uom_ids']
+        return ['id', 'name', 'factor', 'is_pos_groupable']
 
+    def _load_pos_data(self, data):
+        domain = self._load_pos_data_domain(data)
+        fields = self._load_pos_data_fields(data['pos.config'][0]['id'])
+        return self.with_context({**self.env.context}).search_read(domain, fields, load=False)
 
-class UomUom(models.Model):
-    _name = 'uom.uom'
-    _inherit = ['uom.uom', 'pos.load.mixin']
 
-    is_pos_groupable = fields.Boolean(related='category_id.is_pos_groupable', readonly=False)
+class ProductUom(models.Model):
```

Comment:
Should be moved to another file to match your existing `product_uom.py` files, for more consistency/easier searching (uom uom and uom category being in the same file is probably due to it being old code)